linera-faucet/server/src/lib.rs

```diff
@@ -750,3 +734,35 @@ where
         schema.execute(request.into_inner()).await.into()
     }
 }
+
+/// Returns all `(AccountOwner, ChainDescription)` pairs for single-owner chains created in this
+/// block.
+fn extract_opened_single_owner_chains(
+    certificate: &ConfirmedBlockCertificate,
+) -> anyhow::Result<Vec<(AccountOwner, ChainDescription)>> {
```

Comment:
Or make this function more generic, return an iterator over `(AccountOwner, ChainDescription)` and filter out on the call-site all chains that have more than 1 owner?

Reply:
But then we'd have to do the same kind of filtering in all three call sites.